### PR TITLE
docs: add docs for managed EFS

### PR DESCRIPTION
**Pull request by bvtujo**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: site/content/docs/developing/storage.en.md [hunk 1]
```diff
@@ -41,7 +41,105 @@ $ copilot storage init -n my-cluster -t Aurora -w api --engine PostgreSQL --init
...
+  volumes:
+    myManagedEFSVolume:
+      efs: true
+      path: /var/efs
+      read_only: false
+```
+
+This manifest will result in an EFS volume being created at the environment level, with an Access Point and dedicated directory at the path `/frontend` created specifically for your service. This path and EFS volume will persist until you delete your environment. 
+
+The use of an access point for each service ensures that no two services can access each others' data unless you specifically intend for them to do so by specifying the full advanced configuration. You can read more in [Advanced Use Cases](#advanced-use-cases).
```
Comment (huanjani) [inline site/content/docs/developing/storage.en.md]: nit:
```suggestion
The use of an access point for each service ensures that no two services can access each other's data unless you specifically intend for them to do so by specifying the full advanced configuration. You can read more in [Advanced Use Cases](#advanced-use-cases).
```

File: site/content/docs/developing/storage.en.md [hunk 2]
```diff
@@ -41,7 +41,105 @@ $ copilot storage init -n my-cluster -t Aurora -w api --engine PostgreSQL --init
...
+
+When you have populated the directory, modify your manifest to remove the `exec` directive and update the `build` field to your desired Docker build config or image location.
+
+```yaml
+image:
+  build: ./Dockerfile
+storage:
+  volumes:
+    myVolume:
+      efs: enabled
```
Comment (efekarakus) [inline site/content/docs/developing/storage.en.md]:
```suggestion
      efs: true
```

File: site/content/docs/include/common-svc-fields.md [hunk 3]
```diff
@@ -158,15 +158,30 @@ Required. Specify the location in the container where you would like your volume
 <span class="parent-field">volume.</span><a id="read_only" href="#read-only" class="field">`read_only`</a> <span class="type">Bool</span>  
 Optional. Defaults to `true`. Defines whether the volume is read-only or not. If false, the container is granted `elasticfilesystem:ClientWrite` permissions to the filesystem and the volume is writable.
 
-<span class="parent-field">volume.</span><a id="efs" href="#efs" class="field">`efs`</a> <span class="type">Map</span>  
-Specify more detailed EFS configuration.
+<span class="parent-field">volume.</span><a id="efs" href="#efs" class="field">`efs`</a> <span class="type">Bool or Map</span>
```
Comment (huanjani) [inline site/content/docs/include/common-svc-fields.md:161]: nit: for consistency, can we use the full "Boolean" here and elsewhere in the `volumes` fields?

File: site/content/docs/include/common-svc-fields.md [hunk 4]
```diff
@@ -158,15 +158,30 @@ Required. Specify the location in the container where you would like your volume
 <span class="parent-field">volume.</span><a id="read_only" href="#read-only" class="field">`read_only`</a> <span class="type">Bool</span>  
 Optional. Defaults to `true`. Defines whether the volume is read-only or not. If false, the container is granted `elasticfilesystem:ClientWrite` permissions to the filesystem and the volume is writable.
 
-<span class="parent-field">volume.</span><a id="efs" href="#efs" class="field">`efs`</a> <span class="type">Map</span>  
-Specify more detailed EFS configuration.
+<span class="parent-field">volume.</span><a id="efs" href="#efs" class="field">`efs`</a> <span class="type">Bool or Map</span>  
+Specify more detailed EFS configuration. If specified as a Boolean, or using only the `uid` and `gid` subfields, creates a managed EFS filesystem and dedicated Access Point for this workload.
```
Comment (huanjani) [inline site/content/docs/include/common-svc-fields.md]: maybe lowercase?
```suggestion
Specify more detailed EFS configuration. If specified as a boolean, or using only the `uid` and `gid` subfields, creates a managed EFS filesystem and dedicated Access Point for this workload.
```

**Review by huanjani — APPROVED** [review]
Looks great! Clear and informative, especially for a tricky and complicated feature!

**Review by efekarakus — APPROVED** [review]

**Pull request merged**